### DISCUSSION TITLE
Can no longer build if I don't have an iOS device

### DISCUSSION THREAD

**Discussion by GeorgeLeithead** [CommunityToolkit/Maui#874] [General]
When I try to build the latest main branch on a Windows machine, I can no longer get a successful build.

During build I get a dialog asking to set-up a connected iOS device.  I do not HAVE an iOS device, so cancel.  The build then fails, and I cannot perform a successful build.

Build log:

Rebuild started...
Restored CommunityToolKitMaui\src\CommunityToolkit.Maui.Analyzers\CommunityToolkit.Maui.Analyzers.csproj (in 21 ms).
Restored CommunityToolKitMaui\src\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.csproj (in 21 ms).
Restored CommunityToolKitMaui\src\CommunityToolkit.Maui.Analyzers.CodeFixes\CommunityToolkit.Maui.Analyzers.CodeFixes.csproj (in 27 ms).
Restored CommunityToolKitMaui\src\CommunityToolkit.Maui.UnitTests\CommunityToolkit.Maui.UnitTests.csproj (in 197 ms).
Restored CommunityToolKitMaui\src\CommunityToolkit.Maui.Analyzers.UnitTests\CommunityToolkit.Maui.Analyzers.UnitTests.csproj (in 200 ms).
Restored CommunityToolKitMaui\src\CommunityToolkit.Maui.Core\CommunityToolkit.Maui.Core.csproj (in 239 ms).
Restored CommunityToolKitMaui\src\CommunityToolkit.Maui\CommunityToolkit.Maui.csproj (in 239 ms).
Restored CommunityToolKitMaui\samples\CommunityToolkit.Maui.Sample\CommunityToolkit.Maui.Sample.csproj (in 438 ms).
1>------ Rebuild All started: Project: CommunityToolkit.Maui.Analyzers, Configuration: Debug Any CPU ------
2>------ Rebuild All started: Project: CommunityToolkit.Maui.SourceGenerators, Configuration: Debug Any CPU ------
3>------ Rebuild All started: Project: CommunityToolkit.Maui.Core, Configuration: Debug Any CPU ------
1>CommunityToolkit.Maui.Analyzers -> CommunityToolKitMaui\src\CommunityToolkit.Maui.Analyzers\bin\Debug\netstandard2.0\CommunityToolkit.Maui.Analyzers.dll
4>------ Rebuild All started: Project: CommunityToolkit.Maui.Analyzers.CodeFixes, Configuration: Debug Any CPU ------
2>CommunityToolkit.Maui.SourceGenerators -> CommunityToolKitMaui\src\CommunityToolkit.Maui.SourceGenerators\bin\Debug\netstandard2.0\CommunityToolkit.Maui.SourceGenerators.dll
4>CommunityToolkit.Maui.Analyzers.CodeFixes -> CommunityToolKitMaui\src\CommunityToolkit.Maui.Analyzers.CodeFixes\bin\Debug\netstandard2.0\CommunityToolkit.Maui.Analyzers.CodeFixes.dll
5>------ Rebuild All started: Project: CommunityToolkit.Maui.Analyzers.UnitTests, Configuration: Debug Any CPU ------
3>CommunityToolkit.Maui.Core -> CommunityToolKitMaui\src\CommunityToolkit.Maui.Core\bin\Debug\net7.0\CommunityToolkit.Maui.Core.dll
3>CommunityToolkit.Maui.Core -> CommunityToolKitMaui\src\CommunityToolkit.Maui.Core\bin\Debug\net7.0-ios\CommunityToolkit.Maui.Core.dll
3>CommunityToolkit.Maui.Core -> CommunityToolKitMaui\src\CommunityToolkit.Maui.Core\bin\Debug\net7.0-windows10.0.19041.0\CommunityToolkit.Maui.Core.dll
3>CommunityToolkit.Maui.Core -> CommunityToolKitMaui\src\CommunityToolkit.Maui.Core\bin\Debug\net7.0-maccatalyst\CommunityToolkit.Maui.Core.dll
5>CommunityToolkit.Maui.Analyzers.UnitTests -> CommunityToolKitMaui\src\CommunityToolkit.Maui.Analyzers.UnitTests\bin\Debug\net7.0\CommunityToolkit.Maui.Analyzers.UnitTests.dll
3>CommunityToolkit.Maui.Core -> CommunityToolKitMaui\src\CommunityToolkit.Maui.Core\bin\Debug\net7.0-android\CommunityToolkit.Maui.Core.dll
6>------ Rebuild All started: Project: CommunityToolkit.Maui, Configuration: Debug Any CPU ------
6>CommunityToolKitMaui\src\CommunityToolkit.Maui\Behaviors\PlatformBehaviors\IconTintColor\IconTintColorBehavior.shared.cs(8,36,8,66): warning CA1418: The platform 'Tizen' is not a known platform name
6>CommunityToolKitMaui\src\CommunityToolkit.Maui\Behaviors\PlatformBehaviors\StatusBar\StatusBarBehavior.shared.cs(13,106,13,136): warning CA1418: The platform 'Tizen' is not a known platform name
6>CommunityToolkit.Maui -> CommunityToolKitMaui\src\CommunityToolkit.Maui\bin\Debug\net7.0-ios\CommunityToolkit.Maui.dll
6>CommunityToolKitMaui\src\CommunityToolkit.Maui\Behaviors\PlatformBehaviors\IconTintColor\IconTintColorBehavior.shared.cs(8,36,8,66): warning CA1418: The platform 'Tizen' is not a known platform name
6>CommunityToolKitMaui\src\CommunityToolkit.Maui\Behaviors\PlatformBehaviors\StatusBar\StatusBarBehavior.shared.cs(13,106,13,136): warning CA1418: The platform 'Tizen' is not a known platform name
6>CommunityToolkit.Maui -> CommunityToolKitMaui\src\CommunityToolkit.Maui\bin\Debug\net7.0-windows10.0.19041.0\CommunityToolkit.Maui.dll
6>CommunityToolKitMaui\src\CommunityToolkit.Maui\Behaviors\PlatformBehaviors\IconTintColor\IconTintColorBehavior.shared.cs(8,36,8,66): warning CA1418: The platform 'Tizen' is not a known platform name
6>CommunityToolKitMaui\src\CommunityToolkit.Maui\Behaviors\PlatformBehaviors\StatusBar\StatusBarBehavior.shared.cs(13,106,13,136): warning CA1418: The platform 'Tizen' is not a known platform name
6>CommunityToolkit.Maui -> CommunityToolKitMaui\src\CommunityToolkit.Maui\bin\Debug\net7.0\CommunityToolkit.Maui.dll
6>Done building project "CommunityToolkit.Maui.csproj".
6>CommunityToolKitMaui\src\CommunityToolkit.Maui\Behaviors\PlatformBehaviors\IconTintColor\IconTintColorBehavior.shared.cs(8,36,8,66): warning CA1418: The platform 'Tizen' is not a known platform name
6>CommunityToolKitMaui\src\CommunityToolkit.Maui\Behaviors\PlatformBehaviors\StatusBar\StatusBarBehavior.shared.cs(13,106,13,136): warning CA1418: The platform 'Tizen' is not a known platform name
6>CommunityToolkit.Maui -> CommunityToolKitMaui\src\CommunityToolkit.Maui\bin\Debug\net7.0-maccatalyst\CommunityToolkit.Maui.dll
6>Done building project "CommunityToolkit.Maui.csproj".
6>CommunityToolKitMaui\src\CommunityToolkit.Maui\Behaviors\PlatformBehaviors\IconTintColor\IconTintColorBehavior.shared.cs(8,36,8,66): warning CA1418: The platform 'Tizen' is not a known platform name
6>CommunityToolKitMaui\src\CommunityToolkit.Maui\Behaviors\PlatformBehaviors\StatusBar\StatusBarBehavior.shared.cs(13,106,13,136): warning CA1418: The platform 'Tizen' is not a known platform name
6>CommunityToolkit.Maui -> CommunityToolKitMaui\src\CommunityToolkit.Maui\bin\Debug\net7.0-android\CommunityToolkit.Maui.dll
6>Done building project "CommunityToolkit.Maui.csproj".
Verification of iOS environment is running. Please try again in a moment.
The build was cancelled
7>Error: Cancelled
8>------ Rebuild All started: Project: CommunityToolkit.Maui.UnitTests, Configuration: Debug Any CPU ------
8>CommunityToolKitMaui\src\CommunityToolkit.Maui.UnitTests\Behaviors\StatusBarBehaviorTests.cs(12,27,12,50): warning CA1416: This call site is reachable on all platforms. 'StatusBarBehavior' is unsupported on: 'macOS/OSX', 'Windows'.
8>CommunityToolKitMaui\src\CommunityToolkit.Maui.UnitTests\Behaviors\StatusBarBehaviorTests.cs(15,45,15,77): warning CA1416: This call site is reachable on all platforms. 'StatusBarBehavior.StatusBarStyle' is unsupported on: 'macOS/OSX', 'Windows'.
8>CommunityToolKitMaui\src\CommunityToolkit.Maui.UnitTests\Behaviors\StatusBarBehaviorTests.cs(14,36,14,68): warning CA1416: This call site is reachable on all platforms. 'StatusBarBehavior.StatusBarColor' is unsupported on: 'macOS/OSX', 'Windows'.
8>CommunityToolKitMaui\src\CommunityToolkit.Maui.UnitTests\Behaviors\StatusBarBehaviorTests.cs(21,27,21,50): warning CA1416: This call site is reachable on all platforms. 'StatusBarBehavior' is unsupported on: 'macOS/OSX', 'Windows'.
8>CommunityToolKitMaui\src\CommunityToolkit.Maui.UnitTests\Behaviors\StatusBarBehaviorTests.cs(28,3,28,67): warning CA1416: This call site is reachable on all platforms. 'StatusBarBehavior' is unsupported on: 'macOS/OSX', 'Windows'.
8>CommunityToolKitMaui\src\CommunityToolkit.Maui.UnitTests\Behaviors\StatusBarBehaviorTests.cs(28,17,28,66): warning CA1416: This call site is reachable on all platforms. 'StatusBarBehavior' is unsupported on: 'macOS/OSX', 'Windows'.
8>CommunityToolKitMaui\src\CommunityToolkit.Maui.UnitTests\Behaviors\StatusBarBehaviorTests.cs(34,27,34,50): warning CA1416: This call site is reachable on all platforms. 'StatusBarBehavior' is unsupported on: 'macOS/OSX', 'Windows'.
8>CommunityToolkit.Maui.UnitTests -> CommunityToolKitMaui\src\CommunityToolkit.Maui.UnitTests\bin\Debug\net7.0\CommunityToolkit.Maui.UnitTests.dll
8>Done building project "CommunityToolkit.Maui.UnitTests.csproj".
========== Rebuild All: 6 succeeded, 2 failed, 0 skipped ==========
========== Elapsed 00:36.375 ==========

> Rebuild started at 11:01 and took 36.055 seconds

**Comment by TheCodeTraveler** [comment]
Hey George! Have you narrowed down the problem to our code? Or is this a tooling problem or an environment problem specific to your machine?

Our CI builds the project on Windows machines without an iOS Device present. If it were a problem with our code, I imagine the CI Pipeline would fail too (which it hasn't).

**Reply by GeorgeLeithead** [reply]
@brminnick I can confirm that for me, I am often an idiot.  I always use "Windows Machine" and occasionally "Android Emulator" for the target...For what ever reason, when I created a new branch and merged from `main`, it was set to `Local Device` (which I never do, as I don't have an iOS device).  Didn't even occur to me to check that the target was different!

**Reply by TheCodeTraveler** [reply]
Ah - yea, I hate the way that UI Dropdown is designed in Visual Studio.

I've sen many devs hit the same problem when I've been delivering .NET MAUI workshops.

**Comment by CliffAgius** [comment]
Closed and answered
